### DISCUSSION TITLE
Assign new Address to CF2 using USB cable

### DISCUSSION THREAD

**Discussion by magalhaesw** [bitcraze/discussions#928] [Q&A] [answered]
Dear All !
Good Evening !

I'd like to ask for a help.

I am not able to connect to CF2 via Crazyradio-PA because I do not know the ***CF2 address***.

I tried to assign a new address to CF2 using CFclient and USB.
But "write new address to CF2" via USB cable does NOT work. 

The figure below shows the screenshot of CFclient when I assign a new address to CF2.
<img src="https://github.com/bitcraze/discussions/assets/53016613/e52f0d86-556a-447e-b475-89b0fdcb435c" width="300">

&nbsp;
I am not able to flash a new firmware to CF2 using USB cable and CFclient.
<img src="https://github.com/bitcraze/discussions/assets/53016613/dffd6e64-8844-48ad-acde-0a2e7d46ca21" width="450">

&nbsp;
I found ***online*** that I could assign a new address to CF2 by  running the script ***write-eeprom.py***
To assign a new address to CF2 via python script, I followed the instructions available at:
https://www.bitcraze.io/documentation/repository/crazyflie-lib-python/master/development/eeprom/

I downloaded the script ***write-eeprom.py*** from:
https://github.com/bitcraze/crazyflie-lib-python/blob/f35b0baf6ff243d618f7a8becf5a69b1ec6c1f56/examples/memory/write-eeprom.py

When I run the script ***write-eeprom.py***, 
I get the following error:
<img src="https://github.com/bitcraze/discussions/assets/53016613/3d09f35a-834d-447d-8eec-a933ec2c6c61" width="650">

Please,
Could you help me with this issue?
How can I reset the CF2 address to default value?

**Comment by krichardsson** [comment]
You can set the channel and address of the Crazyflie via USB, but note that you have to restart the Crazyflie for the settings to have an effect. 
Flashing can only be done using the radio (not USB).

**Reply by magalhaesw** [reply]
Hi Krichardson!
Thanks for your reply.  

I wrote the new address to CF2 using the USB cable.
But CF2 is not accepting the new address.

Please,
What else can I do to solve this issue?
I'd like to know the present CF2 address or be able to assign a new address.


To change the address via USB cable, 
I read the instructions at: https://github.com/bitcraze/crazyflie-clients-python/issues/310
So, I did as follows,

1) Unplug Crazyradio
2) Start the Crazyflie
3) Connect the Crazyflie with USB cable
4) Start the client
5) "usb://0"  available
6) Click connect
7) CF2 is connected via USB as shown below
<img src="https://github.com/bitcraze/discussions/assets/53016613/0261e5b3-c300-4948-97ac-232c6313de63" width="300">

8) Open the Configure 2.0 dialog
9) Change the address and click write as shown below
<img src="https://github.com/bitcraze/discussions/assets/53016613/44484b81-10ec-4e2c-a47d-73abfd511d62" width="200">

10) Close the configure dialog
11) Unplug USB cable
12) Restart the Crazyflie
13) Plug Crazyradio
14) Start the client
15) Enter the new address in the Address field
16) Click Scan
17) Client does ***NOT*** find the Crazyflie using the new address

**Comment by krichardsson** [comment, marked as answer]
Interesting. 
If you connect with USB again and open the configuration dialog, what address do you see?

If the new address is not set and you still see the old address, you probably need to reflash the NRF  as described in https://github.com/bitcraze/crazyflie-clients-python/issues/310

**Reply by magalhaesw** [reply]
Hi krichardsson,
Thanks for your help.

I am glad to know that the "configuration dialog" provides information about the CF2 when the USB cable is connected.
I was looking for this information, the present address assigned to my CF2. : )

Problem is solved now. 
The issue was that the old address is " 0xE7E7E7E707" which is very close to default value "0xE7E7E7E7E7".

I was trying to connect to "default address" instead of "0xE7E7E7E707".
This was due to lack of attention on my part.

Richardsson,
Your explanation was very helpful to solve this issue. :)) 
 
Thank you !
